include/rocksdb/utilities/debug.h

```diff
@@ -31,9 +31,13 @@ struct KeyVersion {
 };
 
 // Returns listing of all versions of keys in the provided user key range.
-// The range is inclusive-inclusive, i.e., [`begin_key`, `end_key`].
+// The range is inclusive-inclusive, i.e., [`begin_key`, `end_key`], or
+// `max_num_ikeys` has been reached. Since all those keys returned will be
+// copied to memory, if the range covers too many keys, the memory usage
+// may be huge. `max_num_ikeys` can be used to cap the memory usage.
 // The result is inserted into the provided vector, `key_versions`.
 Status GetAllKeyVersions(DB* db, Slice begin_key, Slice end_key,
+                         size_t max_num_ikeys,
```

Comment:
I think max_num_ikeys should be added to the end of the param list with a default value. Its not ideal but would help prevent breaking users code.

Reply:
I thought about it. If it is not for a debug function, I would do that. On the other hand, there is Google C++ Style asking output parameters in the end: https://google.github.io/styleguide/cppguide.html#Output_Parameters

Considering the risk of changing debug function is lower, I am taking a non-compatible API change, and force a limit.

Reply:
Sounds good.